pkg/client/kubernetes/pods.go

```diff
@@ -40,7 +41,16 @@ func GetPodLogs(podInterface corev1client.PodInterface, name string, options *co
 	}
 	defer func() { utilruntime.HandleError(stream.Close()) }()
 
-	return ioutil.ReadAll(stream)
+	logs, err := ioutil.ReadAll(stream)
+	if err != nil {
+		return nil, err
+	}
+
+	if strings.Contains(string(logs), "Internal Error") {
+		return nil, fmt.Errorf("Fetched log contains \"Internal Error\": %q", string(logs))
+	}
+
+	return logs, nil
```

Comment:
I would revert this change now again. IMO it's the duty of the caller to parse the log. This function should only return the log.